packages/extension/src/view/popup/stateProviders/settings/settingsProvider.tsx

```diff
@@ -0,0 +1,180 @@
+
+  const [settingsChanged, setSettingsChanged] = useState<boolean>(false);
+
+  const [isUsingCDP, _setIsUsingCDP] = useState(false);
+  const [isUsingCDPForSettingsDisplay, setIsUsingCDPForSettingsDisplay] =
+    useState(false);
+
+  const intitialSync = useCallback(async () => {
+    const sessionStorage = await chrome.storage.session.get();
+    const currentSettings = await chrome.storage.sync.get();
+
+    if (Object.keys(sessionStorage).includes('pendingReload')) {
+      setSettingsChanged(sessionStorage?.pendingReload);
+
+      if (Object.keys(sessionStorage).includes('allowedNumberOfTabs')) {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          sessionStorage.allowedNumberOfTabs
+        );
+      } else {
+        setAllowedNumberOfTabsForSettingsDisplay(
+          currentSettings.allowedNumberOfTabs
+        );
+      }
+
+      if (Object.keys(sessionStorage).includes('isUsingCDP')) {
+        setIsUsingCDPForSettingsDisplay(sessionStorage.isUsingCDP);
+      } else {
+        setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+      }
+    } else {
+      setAllowedNumberOfTabsForSettingsDisplay(
+        currentSettings.allowedNumberOfTabs
+      );
+      setIsUsingCDPForSettingsDisplay(currentSettings.isUsingCDP);
+    }
+
+    if (Object.keys(currentSettings).includes('allowedNumberOfTabs')) {
+      setAllowedNumberOfTabs(currentSettings.allowedNumberOfTabs);
+    }
+
+    if (Object.keys(currentSettings).includes('isUsingCDP')) {
+      _setIsUsingCDP(currentSettings.isUsingCDP);
+    }
+  }, []);
+
+  const setUsingCDP = useCallback(async (newValue: boolean) => {
+    setIsUsingCDPForSettingsDisplay(newValue);
+    await chrome.storage.session.set({
+      isUsingCDP: newValue,
+      pendingReload: true,
+    });
+  }, []);
+
+  const sessionStoreChangeListener = useCallback(
+    (changes: { [key: string]: chrome.storage.StorageChange }) => {
+      if (
+        Object.keys(changes).includes('allowedNumberOfTabs') &&
+        Object.keys(changes.allowedNumberOfTabs).includes('newValue')
+      ) {
+        setAllowedNumberOfTabs(changes?.allowedNumberOfTabs?.newValue);
+        setSettingsChanged(true);
+      }
+
+      if (
+        Object.keys(changes).includes('isUsingCDP') &&
+        Object.keys(changes.isUsingCDP).includes('newValue')
+      ) {
+        _setIsUsingCDP(changes?.isUsingCDP?.newValue);
+        setSettingsChanged(true);
+      }
+    },
+    []
+  );
+
+  const handleSettingsChange = useCallback(async () => {
+    if (settingsChanged) {
+      await chrome.runtime.sendMessage({
+        type: SERVICE_WORKER_TABS_RELOAD_COMMAND,
+      });
+      setSettingsChanged(false);
+    }
+  }, [settingsChanged]);
+
+  const changeSyncStorageListener = useCallback(
+    (changes: { [key: string]: chrome.storage.StorageChange }) => {
+      if (
+        Object.keys(changes).includes('allowedNumberOfTabs') &&
+        Object.keys(changes.allowedNumberOfTabs).includes('newValue')
```

Comment:
```suggestion
      if (
        changes?.['allowedNumberOfTabs']?.['newValue']
```

Can be used if covers the check, there are other places also in file.

Reply:
Fix in latest commits